### ISSUE TITLE
Item argument not used in Neo.Core.Strings.Count function

### ISSUE THREAD

**Issue by FrankRuben** [AdaDoom3/AdaDoom3#32]
Hi,

the function above as implemented in ~/Engine/neo-core-strings.adb is getting an `Item` argument to count, but instead it's counting `ASCII.CR` instead. If this is intented, it's at least surprising...

Regards, Frank

**Comment by AdaDoom3** [comment]
Hi Frank, thanks for the interest. Investigating this now

**Comment by AdaDoom3** [comment]
You are correct, ASCII.CR should be Item instead. Additionally, the function as a whole is redundant given the existence of Ada.Strings.Wide_Unbounded.Count so the two Count functions should just be completely removed anyway.

I will update this in my next patch when I finally nail down staging of surfaces Vulkan, but if you would like to submit this small change you are certainly welcome to : )

Best regards, Justin

**Comment by FrankRuben** [comment]
Hi Justin,

I gave it a try, but I'm getting various compile errors - using `gprbuild -p -P neo.gpr` - such as:

> neo-api-vulkan.ads:873:12: prefix of "Wide_Image" attribute must be a type
> neo-core-console.adb:213:45: prefix of "Wide_Image" attribute must be a type
> neo-engine-renderer.adb:1596:92: value may be affected by call to "C" because order of evaluation is arbitrary

and my Ada-fu is just good enough to compile simple single-file standalones. Do you have a quick tip? Otherwise I think it would take me an unproportionally larger effort to make that small fix.

Best regards, Frank

**Comment by OneWingedShark** [comment]
> neo-api-vulkan.ads:873:12: prefix of "Wide_Image" attribute must be a type
> neo-core-console.adb:213:45: prefix of "Wide_Image" attribute must be a type

What you probably have here is `Variable'Wide_Image`... the error-message is saying that the attribute "Wide_Image" doesn't apply to a variable, but rather needs a type. (Something like `Integer'Wide_Image`.)

> neo-engine-renderer.adb:1596:92: value may be affected by call to "C" because order of evaluation is arbitrary

What you have here seems to be some possible aliasing, something like `Proc (X,Y : Integer; Z : In Out Integer)` being called with `Proc(C,B,C)` -- or maybe something like `Fn(X : in out Integer) return Integer;` `C:= Fn(C)`.

**Comment by AdaDoom3** [comment]
It appears your compiler is out of date, I think the new GPL compiler should handle the new Ada 2020 'Image attributes.
